### PR TITLE
Stricter CI checks

### PR DESCRIPTION
**Pull request by TheElectronWill**
~~Based on #19 (to review, only look at the last 3 commits).~~

Rebased on main

### CODE REVIEW

**Comment by TheElectronWill** [comment]
Tarpaulin is in trouble because some integration tests run commands... It fails with both backends :S
That will be for another PR.

File: plugin-k8s/src/parsing_cgroupv2.rs [hunk 1]
```diff
@@ -24,8 +24,8 @@ impl FromStr for CgroupV2Metric {
             time_used_user_mode: 0,
             time_used_system_mode: 0,
         };
-        for (_i, line) in s.lines().enumerate() {
```
Comment (titouanj) [inline plugin-k8s/src/parsing_cgroupv2.rs:27]: What was this thing ? 🤣 @AngeCyp

File: Cargo.toml [hunk 2]
```diff
@@ -18,3 +18,14 @@ members = [
...
+# To apply the following lint settings to a workspace member,
+# set "lints.workspace" to true in the member's Cargo.toml.
+[workspace.lints.clippy]
+redundant_field_names = "deny"
+needless_return = "deny"
+needless_borrow = "deny"
+comparison_to_empty = "deny"
+single_char_pattern = "deny"
+unused_enumerate_index = "deny"
+useless_format = "deny"
```
Comment (titouanj) [inline Cargo.toml:31]: Will it put the job in warning state if the is another warning from clippy which is not in this list ?

**Review by titouanj — APPROVED** [review]

**Comment by titouanj** [comment]
> Tarpaulin is in trouble because some integration tests run commands... It fails with both backends :S That will be for another PR.

Doesn't this work ? https://github.com/xd009642/tarpaulin?tab=readme-ov-file#circleci

**Comment by TheElectronWill** [comment]
> > Tarpaulin is in trouble because some integration tests run commands... It fails with both backends :S That will be for another PR.
> 
> Doesn't this work ? https://github.com/xd009642/tarpaulin?tab=readme-ov-file#circleci

It fails in local without the CI

File: Cargo.toml (hunk 2 above)
Comment (TheElectronWill) [reply Cargo.toml:31]: it prints warning in the logs but there's no link with GitHub yet

File: plugin-k8s/src/parsing_cgroupv2.rs (hunk 1 above)
Comment (AngeCyp) [reply plugin-k8s/src/parsing_cgroupv2.rs:27]: I really don't see any problem 🧑‍🦯 :/

**Pull request merged**